Code/Mantid/Framework/MDEvents/src/CoordTransformAffine.cpp

```diff
@@ -41,6 +41,20 @@ CoordTransformAffine::CoordTransformAffine(const size_t inD, const size_t outD)
   // Copy into the raw matrix (for speed)
   copyRawMatrix();
 }
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+ */
+CoordTransformAffine::~CoordTransformAffine() {
+  // delete array of pointers to rows
+  delete[] m_rawMatrix;
+  m_rawMatrix = NULL;
+
+  // delete large mem block holding the matrix
+  delete[] m_rawMemory;
+  m_rawMemory = NULL;
+}
```

Comment:
And a very minor thing: It's unnecessary to set these pointers to `NULL` after freeing them since this is in a destructor so they won't exist any more anyway, but it doesn't do any harm either.